src/expandableCalendar/WeekCalendar/index.tsx

```diff
@@ -1,221 +1,225 @@
+        onDayPress={onDayPressCallback}
+        numberOfDays={numberOfDays}
+        timelineLeftInset={timelineLeftInset}
       />
     );
-  };
+  },[firstDay, onDayPressCallback, context, date]);
 
-  getItemLayout = (_: any, index: number) => {
-    return {
-      length: this.containerWidth,
-      offset: this.containerWidth * index,
-      index
-    };
-  };
-
-  keyExtractor = (_: string, index: number) => index.toString();
+  const keyExtractor = useCallback((item) => item, []);
 
-  renderWeekDaysNames = () => {
+  const renderWeekDaysNames = useMemo(() => {
     return (
-      <WeekDaysNames 
-        firstDay={this.props.firstDay} 
-        style={this.style.dayHeader} 
+      <WeekDaysNames
+        firstDay={firstDay}
+        style={style.current.dayHeader}
       />
     );
-  };
+  },[firstDay]);
 
-  render() {
-    const {allowShadow, firstDay, hideDayNames, current, context} = this.props;
-    const {items} = this.state;
-    const extraData = Map({
-      current,
-      date: context.date,
-      firstDay
-    });
+  const weekCalendarStyle = useMemo(() => {
+    return [
+      allowShadow && style.current.containerShadow,
+      !hideDayNames && style.current.containerWrapper
+    ];
+  }, [allowShadow, hideDayNames]);
 
-    return (
-      <View
-        testID={this.props.testID}
-        style={[allowShadow && this.style.containerShadow, !hideDayNames && this.style.containerWrapper]}
-      >
-        {!hideDayNames && (
-          <View style={[this.style.week, this.style.weekCalendar]}>
-            {/* {this.props.weekNumbers && <Text allowFontScaling={false} style={this.style.dayHeader}></Text>} */}
-            {this.renderWeekDaysNames()}
-          </View>
-        )}
-        <FlatList
-          ref={this.presenter.list}
-          data={items}
-          extraData={extraData}
-          style={this.style.container}
-          horizontal
-          showsHorizontalScrollIndicator={false}
-          pagingEnabled
-          scrollEnabled
-          renderItem={this.renderItem}
-          keyExtractor={this.keyExtractor}
-          initialScrollIndex={NUMBER_OF_PAGES}
-          getItemLayout={this.getItemLayout}
-          onScroll={this.onScroll}
-          onMomentumScrollEnd={this.onMomentumScrollEnd}
-        />
+  const containerStyle = useMemo(() => {
+    return [style.current.week, style.current.weekCalendar];
+  }, []);
+
+  const flatListContainerStyle = useMemo(() => {
+    return style.current.container;
+  }, []);
+
+  const getItemLayout = useCallback((_, index: number) => {
+    return {
+      length: containerWidth,
+      offset: containerWidth * index,
+      index
+    };
+  }, [containerWidth]);
+
+  const onViewableItemsChanged = useCallback(({viewableItems}: { viewableItems: Array<ViewToken>}) => {
+    const newDate = viewableItems[0]?.item;
+    if (changedItems.current) {
+      changedItems.current = false;
+      return;
+    }
+    if (newDate !== visibleWeek.current) {
```

Comment:
Don't you need to check if the two dates are not in the same week before updating?

Reply:
newDate is the item name/date (since this is function is only for week scroll), since each item is a different week then by definition they are not in the same week

Reply:
Sure, only that `viewableItems` is an array that can contain the current week (depends on the viewability config), and since the weeks are actually marked as day dates it is possible that the `newDate` is different from the `visibleWeek.current` (One is Sunday and the other Wednesday) and still on the same week.

Reply:
i tested it many times, using sameWeek will only cause issues when passing number of days (as one week can have 2/3 different items

Reply:
oh ok. Got it. Thanks